rs/ic_os/os_tools/hostos_tool/src/main.rs

```diff
@@ -56,78 +56,116 @@ pub fn main() -> Result<()> {
+                        mgmt_mac.get()
+                    );
+                    mgmt_mac
+                }
+                None => get_ipmi_mac()?,
+            };
+            let generated_mac = generate_mac_address(
+                &mgmt_mac,
+                deployment_settings.deployment.name.as_str(),
                 &node_type,
-                deployment_settings.deployment.mgmt_mac.as_deref(),
             )?;
-            let ipv6_address = generate_ipv6_address(&network_info.ipv6_prefix, &mac)?;
+            let ipv6_address = generate_ipv6_address(&network_info.ipv6_prefix, &generated_mac)?;
             println!("{}", to_cidr(ipv6_address, network_info.ipv6_subnet));
             Ok(())
         }
         Some(Commands::GenerateMacAddress { node_type }) => {
-            let config_map = config_map_from_path(Path::new(&opts.config))
-                .context("Please specify a valid config file with '--config'")?;
+            let config_map = config_map_from_path(Path::new(&opts.config)).context(format!(
+                "Failed to get config.ini settings for path: {}",
+                &opts.config
+            ))?;
             eprintln!("Using config: {:?}", config_map);
 
             let network_info = NetworkInfo::from_config_map(&config_map)?;
             eprintln!("Network info config: {:?}", &network_info);
 
             let deployment_settings = get_deployment_settings(Path::new(&opts.deployment_file))
-                .context("Please specify a valid deployment file with '--deployment-file'")?;
+                .context(format!(
+                    "Failed to get deployment settings for file: {}",
+                    &opts.deployment_file
+                ))?;
             eprintln!("Deployment config: {:?}", deployment_settings);
 
             let node_type = node_type.parse::<NodeType>()?;
-            let mac = generate_mac_address(
-                &deployment_settings.deployment.name,
+            let mgmt_mac = match deployment_settings.deployment.mgmt_mac {
+                Some(config_mac) => {
+                    let mgmt_mac = FormattedMacAddress::try_from(config_mac.as_str())?;
+                    eprintln!(
+                        "Using mgmt_mac address found in deployment.json: {}",
+                        mgmt_mac.get()
+                    );
+                    mgmt_mac
+                }
+                None => get_ipmi_mac()?,
+            };
+            let generated_mac = generate_mac_address(
+                &mgmt_mac,
+                deployment_settings.deployment.name.as_str(),
                 &node_type,
-                deployment_settings.deployment.mgmt_mac.as_deref(),
             )?;
-            let mac = FormattedMacAddress::from(&mac);
-            println!("{}", mac.get());
+
+            let generated_mac = FormattedMacAddress::from(&generated_mac);
+            println!("{}", generated_mac.get());
```

Comment:
Does this work printing directly?

Reply:
It's used in production so it must be working:
`$ rg generate-mac-address
ic-os/components/hostos-scripts/generate-guestos-config/generate-guestos-config.sh
109:    MAC_ADDRESS=$(/opt/ic/bin/hostos_tool generate-mac-address --node-type GuestOS)

ic-os/components/hostos-scripts/generate-guestos-config/dev-generate-guestos-config.sh
120:    MAC_ADDRESS=$(/opt/ic/bin/hostos_tool generate-mac-address --node-type GuestOS)`